src/features/targeting/TargetingApi.js

```diff
@@ -4,11 +4,15 @@
  * basis. Currently, useMessage can provide targeting from MessageContext.
  */
 
+import type { MessageEvent } from './types';
+
 export type TargetingApi = $ReadOnly<{
     // is true if message is eligible to show and not yet closed
-    canShow: boolean,
-    // call onClose just before message should be closed, to cause canShow to be false
+    canShow: () => boolean,
```

Comment:
convert canShow back to function as in a State Machine situation such as ftux, the canShow comes from a function injected by state machine